src/Sylius/Behat/Context/Setup/OrderContext.php

```diff
@@ -275,7 +275,27 @@ public function theCustomerChoseShippingMethod(ShippingMethodInterface $shipping
 
         $this->applyTransitionOnOrderCheckout($order, OrderCheckoutTransitions::TRANSITION_SELECT_SHIPPING);
         $this->applyTransitionOnOrderCheckout($order, OrderCheckoutTransitions::TRANSITION_COMPLETE);
-        $this->stateMachineFactory->get($order, OrderPaymentTransitions::GRAPH)->apply(OrderPaymentTransitions::TRANSITION_PAY);
+        if (!$order->getPayments()->isEmpty()) {
+            $this->stateMachineFactory->get($order, OrderPaymentTransitions::GRAPH)->apply(OrderPaymentTransitions::TRANSITION_PAY);
+        }
+
+        $this->objectManager->flush();
+    }
+
+    /**
+     * @Given /^the customer chose ("[^"]+" payment)$/
+     */
+    public function theCustomerChosePayment(PaymentMethodInterface $paymentMethod)
+    {
+        /** @var OrderInterface $order */
+        $order = $this->sharedStorage->get('order');
+
+        foreach ($order->getPayments() as $payment) {
+            $payment->setMethod($paymentMethod);
+        }
+
+        $this->applyTransitionOnOrderCheckout($order, OrderCheckoutTransitions::TRANSITION_SELECT_PAYMENT);
+        $this->applyTransitionOnOrderCheckout($order, OrderCheckoutTransitions::TRANSITION_COMPLETE);
```

Comment:
Should it be a part of this step?

Reply:
It's a similar implementation to the method `theCustomerChoseShippingMethod`. I totally agree we should have two separate states for that, but to make it consistent I would merge it as it is, to not include more unrelated changes in this PR.

Reply:
just asking :)